packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -1005,17 +1004,17 @@ class FlutterError extends Error with DiagnosticableTreeMixin implements Asserti
       final String package = '${frame.packageScheme}:${frame.package}';
       if (removedPackagesAndClasses.containsKey(className)) {
         skipped += 1;
-        removedPackagesAndClasses[className] += 1;
+        removedPackagesAndClasses.update(className, (int value) => value + 1);
```

Comment:
Why is this change necessary?

Reply:
`someMap[someKey]` is optional.

Reply:
Without this change there's an error because `removedPackagesAndClasses[className]` is nullable and you cannot use operator + on a nullable value. Without `.update` we should have used:

```dart
removedPackagesAndClasses[className] = removedPackagesAndClasses[className]! + 1;
```

The update version avoids repetitions and is less error prone imho.

Reply:
Agreed, `update` does look nicer. Thanks for the explanation.